components/brave_gcm_driver/brave_gcm_channel_status_syncer.cc

```diff
@@ -0,0 +1,54 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_gcm_driver/brave_gcm_channel_status_syncer.h"
+
+#include "base/memory/ptr_util.h"
+#include "chrome/browser/profiles/profile.h"
```

Comment:
you can't have dependencies on `chrome/browser` inside a component. This class should be moved to `brave/browser/gcm_driver/`

Reply:
actually that's a problem because of this https://github.com/brave/brave-core/pull/3390/files#diff-b7494bf5a7e56428c188f4349d20eb49R10

Reply:
do we still need ^^

Reply:
yes we are capturing the initial pref status using the subclass `BraveGCMChannelStatusSyncer`

Reply:
> you can't have dependencies on `chrome/browser` inside a component. This class should be moved to `brave/browser/gcm_driver/`

will move.

Reply:
it seems like we don't need the `BraveGCMChannelStatusSyncer` class at all anymore and only need BraveGCMChannelStatus?